src/test/scala/cognite/spark/EventsRelationTest.scala

```diff
@@ -64,6 +64,22 @@ class EventsRelationTest extends FlatSpec with Matchers with SparkTest {
     assert(eventsRead == 8)
   }
 
+  it should "handle duplicates in a pushdown filter scenario" taggedAs WriteTest in {
+    val metricsPrefix = "single.pushdown.filter.duplicates"
+    val df = spark.read
+      .format("cognite.spark")
+      .option("apiKey", writeApiKey.apiKey)
+      .option("type", "events")
+      .option("collectMetrics", "true")
+      .option("metricsPrefix", metricsPrefix)
+      .option("partitions", "1")
```

Comment:
Try this with several partitions.

Reply:
Good call, discovered a bug in the implementation when I tried higher values.